pkg/services/mto_shipment/mto_shipment_updater.go

```diff
@@ -1282,7 +1282,10 @@ func CalculateRequiredDeliveryDate(appCtx appcontext.AppContext, planner route.P
 			}
 		} else {
 			if intlTransTime.UbTransitTime != nil {
-				requiredDeliveryDate = requiredDeliveryDate.AddDate(0, 0, *intlTransTime.UbTransitTime)
+				if intlTransTime.UbTransitTime != nil {
+					dayAfterPickupDate := pickupDate.AddDate(0, 0, 1)
+					requiredDeliveryDate = dayAfterPickupDate.AddDate(0, 0, *intlTransTime.UbTransitTime)
+				}
```

Comment:
```suggestion
					dayAfterPickupDate := pickupDate.AddDate(0, 0, 1)
					requiredDeliveryDate = dayAfterPickupDate.AddDate(0, 0, *intlTransTime.UbTransitTime)
```

Current code double dips on `intlTransTime.UbTransitTime != nil`

Reply:
@cameroncaci  unintended extra nil check removed

Reply:
Resolving comment